### PR TITLE
LibGfx: Some more path rasterizer tweaks 

### PR DESCRIPTION
**Pull request by MacDue**
[screen-capture - 2024-01-06T015707.131.webm](https://github.com/SerenityOS/serenity/assets/11597044/5d29733f-43a6-40dd-8fae-60b070c131bc)

Collection of small tweaks/hacks, this improves performance a bit more over #22547.

### CODE REVIEW

File: Userland/Libraries/LibGfx/EdgeFlagPathRasterizer.cpp [hunk 1]
```diff
@@ -96,8 +96,6 @@ template<unsigned SamplesPerPixel>
 EdgeFlagPathRasterizer<SamplesPerPixel>::EdgeFlagPathRasterizer(IntSize size)
     : m_size(size.width() + 1, size.height() + 1)
 {
-    // FIXME: Clip the scanline width to the visible section (tricky).
```
Comment (nico) [inline Userland/Libraries/LibGfx/EdgeFlagPathRasterizer.cpp:99]: Commit message: s/passed/past/ (...I think?)

Comment (MacDue) [reply Userland/Libraries/LibGfx/EdgeFlagPathRasterizer.cpp:99]: Fixed

**Comment by awesomekling** [comment]
This crashes on macOS CI:

```
/Users/runner/work/1/s/Userland/Libraries/LibGfx/Path.cpp:519:26: runtime error: nan is outside the range of representable values of type 'int'
    #0 0x10f153e83 in Gfx::Path::stroke_to_fill(float) const Path.cpp:519
    #1 0x10ec26875 in Gfx::AntiAliasingPainter::stroke_path(Gfx::Path const&, Gfx::Color, float) AntiAliasingPainter.cpp:200
    #2 0x11aebc435 in Web::Painting::PaintingCommandExecutorCPU::stroke_path_using_color(Gfx::Path const&, Gfx::Color const&, float, Gfx::Point<float> const&) PaintingCommandExecutorCPU.cpp:293
    #3 0x11aeda5d4 in Web::Painting::RecordingPainter::execute(Web::Painting::PaintingCommandExecutor&) RecordingPainter.cpp:489
    #4 0x11b05c4c8 in Web::SVG::SVGDecodedImageData::render(Gfx::Size<int>) const SVGDecodedImageData.cpp:138
    #5 0x11b05d065 in Web::SVG::SVGDecodedImageData::bitmap(unsigned long, Gfx::Size<int>) const SVGDecodedImageData.cpp:151
    #6 0x11a0db4d6 in Web::CSS::ImageStyleValue::bitmap(unsigned long, Gfx::Size<int>) const ImageStyleValue.cpp:97
    #7 0x11a0dc74f in Web::CSS::ImageStyleValue::paint(Web::PaintContext&, Gfx::Rect<AK::DistinctNumeric<int, Web::__DevicePixels_tag, AK::DistinctNumericFeature::Arithmetic, AK::DistinctNumericFeature::CastToUnderlying, AK::DistinctNumericFeature::Comparison, AK::DistinctNumericFeature::Increment> > const&, Web::CSS::ImageRendering) const ImageStyleValue.cpp:136
    #8 0x11ae36bd4 in Web::Painting::paint_background(Web::PaintContext&, Web::Layout::NodeWithStyleAndBoxModelMetrics const&, Gfx::Rect<Web::CSSPixels> const&, Gfx::Color, Web::CSS::ImageRendering, AK::Vector<Web::CSS::BackgroundLayerData, 0ul> const*, Web::Painting::BorderRadiiData const&) BackgroundPainting.cpp:393
    #9 0x11aea3a33 in Web::Painting::PaintableBox::paint_background(Web::PaintContext&) const PaintableBox.cpp:334
    #10 0x11ae9e737 in Web::Painting::PaintableBox::paint(Web::PaintContext&, Web::Painting::PaintPhase) const PaintableBox.cpp:196
    #11 0x11aeada29 in Web::Painting::PaintableWithLines::paint(Web::PaintContext&, Web::Painting::PaintPhase) const PaintableBox.cpp:626
    #12 0x11af0fd54 in Web::Painting::paint_node(Web::Painting::Paintable const&, Web::PaintContext&, Web::Painting::PaintPhase) StackingContext.cpp:30
    #13 0x11af1d360 in auto Web::Painting::StackingContext::paint_descendants(Web::PaintContext&, Web::Painting::Paintable const&, Web::Painting::StackingContext::StackingContextPaintPhase)::$_1::operator()<Web::Painting::Paintable>(Web::Painting::Paintable&) const StackingContext.cpp:132
    #14 0x11af101d0 in Web::Painting::StackingContext::paint_descendants(Web::PaintContext&, Web::Painting::Paintable const&, Web::Painting::StackingContext::StackingContextPaintPhase) StackingContext.cpp:97
    #15 0x11af1dcfd in auto Web::Painting::StackingContext::paint_descendants(Web::PaintContext&, Web::Painting::Paintable const&, Web::Painting::StackingContext::StackingContextPaintPhase)::$_1::operator()<Web::Painting::Paintable>(Web::Painting::Paintable&) const StackingContext.cpp:136
    #16 0x11af101d0 in Web::Painting::StackingContext::paint_descendants(Web::PaintContext&, Web::Painting::Paintable const&, Web::Painting::StackingContext::StackingContextPaintPhase) StackingContext.cpp:97
    #17 0x11af1301e in Web::Painting::StackingContext::paint_internal(Web::PaintContext&) const StackingContext.cpp:213
    #18 0x11af11f6a in Web::Painting::StackingContext::paint(Web::PaintContext&) const StackingContext.cpp:373
    #19 0x11af10948 in Web::Painting::StackingContext::paint_child(Web::PaintContext&, Web::Painting::StackingContext const&) StackingContext.cpp:188
    #20 0x11af1d195 in auto Web::Painting::StackingContext::paint_descendants(Web::PaintContext&, Web::Painting::Paintable const&, Web::Painting::StackingContext::StackingContextPaintPhase)::$_1::operator()<Web::Painting::Paintable>(Web::Painting::Paintable&) const StackingContext.cpp:122
    #21 0x11af101d0 in Web::Painting::StackingContext::paint_descendants(Web::PaintContext&, Web::Painting::Paintable const&, Web::Painting::StackingContext::StackingContextPaintPhase) StackingContext.cpp:97

SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior /Users/runner/work/1/s/Userland/Libraries/LibGfx/Path.cpp:519:26 in 
WebContent Crashed!!
    Last started test: /Users/runner/work/1/s/Tests/LibWeb/Ref/svg-background-no-natural-size.html
```

**Comment by MacDue** [comment]
Should be fixed now :thinking:

**Pull request merged**